app/models/host/managed.rb

```diff
@@ -985,7 +995,7 @@ def ensure_associations
   end
 
   def set_default_user
-    return if self.owner_type.present? && !OWNER_TYPES.include?(self.owner_type)
+    return if self.owner_type.present? && !OWNER_TYPES.include?(self.owner_type) # to propagate invalid value to validations
     self.owner_type ||= 'User'
```

Comment:
After #3740 was merged I modified this so the set_default_user only sets default user if both are missing, ignoring OWNER_TYPES as it's handled by `Host::Base` validation. Now user can't create host without specifying either both or none, hammer example:

```sh
# current user will be used
$ hammer host create --managed false --name a --location Brno --organization Ares --build false
Host created

$ hammer host create --managed false --name a --location Brno --organization Ares --build false --owner-type Usergroup
Could not create the host:
  Owner If owner type is specified, owner must be specified too.
$ hammer host create --managed false --name a --location Brno --organization Ares --build false --owner-type User     
Could not create the host:
  Owner If owner type is specified, owner must be specified too.

$ hammer host create --managed false --name a --location Brno --organization Ares --build false --owner-id 2     
Could not create the host:
  Owner type Owner type needs to be one of the following: User, Usergroup

$ hammer host create --managed false --name a --location Brno --organization Ares --build false --owner-id 2 --owner-type User
Host created
```